android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/CatapultAppcoinsBilling.java

```diff
@@ -30,111 +28,119 @@
+        int responseCode;
+
+        try {
+            WalletUtils.getSdkAnalytics().sendPurchaseIntentEvent(billingFlowParams.getSku());
+            String payload = PayloadHelper.buildIntentPayload(billingFlowParams.getOrderReference(),
+                    billingFlowParams.getDeveloperPayload(), billingFlowParams.getOrigin());
+            AttributionSharedPreferences attributionSharedPreferences =
+                    new AttributionSharedPreferences(activity);
+            String oemid = attributionSharedPreferences.getOemId();
+            String guestWalletId = attributionSharedPreferences.getWalletId();
 
-  @Override public int launchBillingFlow(Activity activity, BillingFlowParams billingFlowParams) {
+            logDebug("Message: " + payload);
 
-    int responseCode;
+            Thread eventLoggerThread = new Thread(new EventLogger(billingFlowParams.getSku(),
+                    activity.getApplicationContext()
+                            .getPackageName()));
+            eventLoggerThread.start();
 
-    try {
-      WalletUtils.getSdkAnalytics().sendPurchaseIntentEvent(billingFlowParams.getSku());
-      String payload = PayloadHelper.buildIntentPayload(billingFlowParams.getOrderReference(),
-          billingFlowParams.getDeveloperPayload(), billingFlowParams.getOrigin());
-      AttributionSharedPreferences attributionSharedPreferences =
-          new AttributionSharedPreferences(activity);
-      String oemid = attributionSharedPreferences.getOemId();
-      String guestWalletId = attributionSharedPreferences.getWalletId();
+            LaunchBillingFlowResult launchBillingFlowResult =
+                    billing.launchBillingFlow(billingFlowParams, payload, oemid, guestWalletId);
 
-      logDebug("Message: " + payload);
+            responseCode = launchBillingFlowResult.getResponseCode();
 
-      Thread eventLoggerThread = new Thread(new EventLogger(billingFlowParams.getSku(),
-          activity.getApplicationContext()
-              .getPackageName()));
-      eventLoggerThread.start();
+            if (responseCode != ResponseCode.OK.getValue()) {
+                ApplicationUtils.handleActivityResult(
+                        billing,
+                        ResponseCode.ERROR.getValue(),
+                        null,
+                        purchaseFinishedListener
+                );
+                return responseCode;
+            }
 
-      LaunchBillingFlowResult launchBillingFlowResult =
-          billing.launchBillingFlow(billingFlowParams, payload, oemid, guestWalletId);
+            PendingIntent buyIntent = launchBillingFlowResult.getBuyIntent();
+            Intent webBuyIntent = launchBillingFlowResult.getWebBuyIntent();
 
-      responseCode = launchBillingFlowResult.getResponseCode();
+            PaymentsResultsManager.getInstance().collectPaymentResult(this);
+            if (buyIntent != null) {
+                activity.startActivity(BillingFlowActivity.newIntent(activity, buyIntent));
```

Comment:
Improve this by creating the Intent in the `WalletBillingService` and use only `BUY_INTENT`.

Reply:
Done.